### PR TITLE
Add the support for custom name section

### PR DESCRIPTION
**Pull request by qinxk-inter**

### CODE REVIEW

File: core/iwasm/interpreter/wasm_mini_loader.c [hunk 1]
```diff
@@ -1407,6 +1407,70 @@ load_start_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
...
+    while (p < p_end) {
+        read_leb_uint32(p, p_end, name_type);
+        if (i != 0) {
+            bh_assert(name_type > previous_name_type);
+        }
+        previous_name_type = name_type;
+        read_leb_uint32(p, p_end, subsection_size);
+        CHECK_BUF(p, p_end, subsection_size);
+        switch (name_type)
+        {
```
Comment (xujuntwt95329) [inline core/iwasm/interpreter/wasm_mini_loader.c]: same as above

File: core/iwasm/interpreter/wasm_loader.c [hunk 2]
```diff
@@ -2350,6 +2350,91 @@ load_start_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
...
+            if (name_type < previous_name_type ) {
+                set_error_buf(error_buf, error_buf_size, "out-of-order sub-section");
+                return false;
+            }
+        }
+        previous_name_type = name_type;
+        read_leb_uint32(p, p_end, subsection_size);
+        CHECK_BUF(p, p_end, subsection_size);
+        switch (name_type)
+        {
```
Comment (xujuntwt95329) [inline core/iwasm/interpreter/wasm_loader.c]: had better move `{` to last line
```suggestion
        switch (name_type) {
```

File: build-scripts/config_common.cmake [hunk 3]
```diff
@@ -165,4 +165,7 @@ endif ()
 if (DEFINED WAMR_APP_THREAD_STACK_SIZE_MAX)
   add_definitions (-DAPP_THREAD_STACK_SIZE_MAX=${WAMR_APP_THREAD_STACK_SIZE_MAX})
 endif ()
+if (WAMR_BUILD_CUSTOM_NAME_SECTION EQUAL 1)
+  add_definitions (-DWASM_ENABLE_CUSTOM_NAME_SECTION=1)
```
Comment (xujuntwt95329) [inline build-scripts/config_common.cmake:173]: Had better print a message to indicate the feature is enabled
```suggestion
  add_definitions (-DWASM_ENABLE_CUSTOM_NAME_SECTION=1)
  message ("     Custom name section enabled")
```

File: core/iwasm/interpreter/wasm_loader.c [hunk 4]
```diff
@@ -2350,6 +2350,101 @@ load_start_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
     return false;
 }
 
+#if WASM_ENABLE_CUSTOM_NAME_SECTION != 0
+static bool
+handle_name_section(const uint8 *buf,
+                    const uint8 *buf_end,
+                    WASMModule *module,
+                    char *error_buf,
+                    uint32 error_buf_size)
```
Comment (wenyongh) [inline core/iwasm/interpreter/wasm_loader.c]: Could we change to:
```
static bool
handle_name_section(const uint8 *buf, const uint8 *buf_end,
                    WASMModule *module,
                    char *error_buf, uint32 error_buf_size)
```

File: core/iwasm/interpreter/wasm_mini_loader.c [hunk 5]
```diff
@@ -1407,6 +1407,75 @@ load_start_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
     return true;
 }
 
+#if WASM_ENABLE_CUSTOM_NAME_SECTION != 0
+static bool
+handle_name_section(const uint8 *buf,
+                    const uint8 *buf_end,
+                    WASMModule *module,
+                    char *error_buf,
+                    uint32 error_buf_size)
```
Comment (wenyongh) [inline core/iwasm/interpreter/wasm_mini_loader.c]: Same as above

**Comment by qinxk-inter** [comment]
All review comments have been addressed and also the latest PR has been rebased on latest main branch.

**Pull request merged**